osm2lanes/src/overpass.rs

```diff
@@ -171,36 +185,128 @@ pub async fn get_nearby(
                 area.enclosing["ISO3166-1"];
                 area.enclosing["driving_side"];
             );
-            out tags;"#,
-            lng_lat.0, lng_lat.1
+            out tags;"#
         ))
         .send()
         .await?
         .json::<OverpassResult>()
         .await?;
     log::debug!("{:#?}", resp);
-    if resp
-        .elements
-        .iter()
-        .filter(|e| e.geometry.is_some())
-        .count()
-        > 1
-    {
-        log::warn!("more than one nearby way found, returning one at random");
-    }
 
     let locale = resp.locale();
-
-    resp.elements.truncate(1);
-    let element = resp.elements.pop().unwrap();
+    let (element, geometry, _distance) = resp
+        .elements
+        .into_iter()
+        .filter_map(|element| {
+            let geometry = element.geometry.as_ref().map(|g| convert(g));
+            geometry.map(|geometry| {
+                let distance = point.euclidean_distance(&geometry);
+                (element, geometry, distance)
+            })
+        })
+        .min_by(|(_, _, left_distance), (_, _, right_distance)| {
```

Comment:
yeah, handling floats is annoying.
I wish there was a zero-cost never-nan float type in rust's std that could be used instead. Apparently the discussions on that stall because of how hard it is to make a good API.

Something like https://docs.rs/decorum/latest/decorum/type.R64.html looks like a nice solution.